x-pack/plugins/security_solution/public/common/hooks/eql/use_eql_preview.ts

```diff
@@ -3,10 +3,157 @@
 
-const getEqlPreviewWithOptionalSignal = withOptionalSignal(getEqlPreview);
+export const useEqlPreview = (): [
+  boolean,
+  (arg: EqlPreviewRequest) => void,
+  EqlPreviewResponse
+] => {
+  const { data, notifications } = useKibana().services;
+  const refetch = useRef<inputsModel.Refetch>(noop);
+  const abortCtrl = useRef(new AbortController());
+  const unsubscribeStream = useRef(new Subject());
+  const [loading, setLoading] = useState(false);
+  const didCancel = useRef(false);
 
-export const useEqlPreview = () => useAsync(getEqlPreviewWithOptionalSignal);
+  const [response, setResponse] = useState<EqlPreviewResponse>({
+    data: [],
+    inspect: {
+      dsl: [],
+      response: [],
+    },
+    refetch: refetch.current,
+    totalCount: 0,
+  });
+
+  const searchEql = useCallback(
+    ({ from, to, query, index, interval }: EqlPreviewRequest) => {
+      if (parseScheduleDates(to) == null || parseScheduleDates(from) == null) {
+        notifications.toasts.addWarning('Time intervals are not defined.');
+        return;
+      }
+
+      const asyncSearch = async () => {
+        abortCtrl.current = new AbortController();
+        setLoading(true);
+        setResponse((prevResponse) => ({
+          ...prevResponse,
+          data: [],
+          inspect: {
+            dsl: [],
+            response: [],
+          },
+          totalCount: 0,
+        }));
+
+        data.search
+          .search<EqlSearchStrategyRequest, EqlSearchStrategyResponse<EqlSearchResponse<Source>>>(
+            {
+              params: {
+                // @ts-expect-error allow_no_indices is missing on EqlSearch
+                allow_no_indices: true,
+                index: index.join(),
+                body: {
+                  filter: {
+                    range: {
+                      '@timestamp': {
+                        gte: from,
+                        lte: to,
+                        format: 'strict_date_optional_time',
+                      },
+                    },
+                  },
+                  query,
+                  // EQL requires a cap, otherwise it defaults to 10
+                  // It also sorts on ascending order, capping it at
+                  // something smaller like 20, made it so that some of
+                  // the more recent events weren't returned
+                  size: 100,
+                },
+              },
+            },
+            {
+              strategy: EQL_SEARCH_STRATEGY,
+              abortSignal: abortCtrl.current.signal,
+            }
+          )
+          .pipe(takeUntil(unsubscribeStream.current))
+          .subscribe({
+            next: (res) => {
+              if (isCompleteResponse(res)) {
+                if (!didCancel.current) {
+                  setLoading(false);
+                  if (hasEqlSequenceQuery(query)) {
+                    setResponse(getSequenceAggs(res, refetch.current));
```

Comment:
I think `getSequenceAggs` is the source of these console warnings:

<img width="830" alt="DevTools_-_localhost_5601_app_security_detections_rules_create_sourcerer__default______timerange__global__linkTo___timeline__timerange__from__272020-10-13T14_40_11_422Z_27_fromStr_now-30m_kind_relative_to__272020-10-13T15_10_11_422Z_27_toSt" src="https://user-images.githubusercontent.com/657252/95885670-28002480-0d43-11eb-9e38-6170f3f2aba2.png">

Since the x-values are time strings and not numbers. However, wrapping them in `moment().valueOf()` fixes the warning but breaks the "buckets", since the x-axis is now back to using a linear scale:

<img width="913" alt="Detections_-_Kibana" src="https://user-images.githubusercontent.com/657252/95886465-3b5fbf80-0d44-11eb-840b-2028283845ba.png">

Reply:
Could we just preview this in an embedded Resolver?

Reply:
I'm not sure what you have against rainbow graphs @rylnd  😜  I have a draft where I'm trying to see if displaying it in more of a Gantt type form helps. Will definitely follow up with @MikePaquette and @marrasherrier on this.

@madirey I wonder how heavy that would be, or is it already easily embeddable?